File: java/src/Polimorfismo/Cone.java
package Polimorfismo;

public class Cone extends Figuras{

    Cone(String nome, double dim1, double dim2, double dim3) {
        this.nome = nome;
        this.dim1 = dim1;
        this.dim2 = dim2;
        this.dim3 = dim3;
    }

    public double volume(double dim1, double dim3) {
        double volume;
        volume = 1 / 3 * Math.PI * Math.pow(dim1, 2) * dim3;
        return volume;
    }

    public double area(double dim1, double dim2) {
        double area;
        area = (Math.PI * Math.pow(dim1, 2)) + (Math.PI * dim1 * dim2);
        return area;
    }
    void display(){
        System.out.println(nome + dim1 + dim2 + dim3);
    }
}

File: cpp/Polimorfismo/Cone.cpp
#include <cmath>
#include <iostream>
#include "Figuras.h"

class Cone : public Figuras {
public:
    Cone(std::string nome, double dim1, double dim2, double dim3) : Figuras(nome, dim1, dim2, dim3) {}

    double volume() override {
        return (1.0 / 3) * M_PI * std::pow(dim1, 2) * dim3;
    }

    double area() override {
        return M_PI * std::pow(dim1, 2) + M_PI * dim1 * dim2;
    }

    void display() override {
        std::cout << nome << ": Raio = " << dim1 << ", Altura = " << dim2 << ", Geratriz = " << dim3 << std::endl;
    }

    void aumentar() override {
        dim1 *= aumento;
        dim2 *= aumento;
        dim3 *= aumento;
    }
};